### ISSUE TITLE
Hide password by default from formatted printing

### ISSUE THREAD

**Issue by lo48576** [lo48576/iri-string#25]
> Applications should not render as clear text any data after the first colon (":") character found within a userinfo subcomponent unless the data after the colon is the empty string (indicating no password).
>
> --- [RFC 3986 section 3.2.1. User Information](https://www.rfc-editor.org/rfc/rfc3986.html#section-3.2.1)

According to this, `println!("{}", uri);` or such "usual" formatting should hide the password part of the `userinfo` component, unless explicitly specified.
To see full userinfo, `uri.display_unsecure()` or `uri.display_secrets()` would have to be provided.

**Comment by lo48576** [comment]
This doesn't change API breakingly (i.e. downstream code compilable with the previous verisons are still compilable after this change), but this will break some kind of codes semantically.
Because of this, I treat this as a breaking change.

**Comment by lo48576** [comment]
Is it enough if `Display` writes `https://user@example.com/` and `Debug` writes `https://user:pass@example.com/`?
I think this may still have a risk that the user unintentionally prints sensitive information into logging file, error page, or something like that.

**Comment by lo48576** [comment]
I decided not to hide the password by default.
Newly added  `{BorrowedIri}::mask_password()` method lets users hide or replace the password part.

The reason not to change the default is, some types requires `core::fmt::Display` to write the raw IRI including the password part, since strings generation by builder and proxy types are built on `core::fmt::Display` trait.
If some types hides the password and some others don't, it will confuse users and may become the cause of the vulnerability.
So I decided to let `core::fmt::Display` write the raw (uncensored) values anytime, and require users to explicitly mask IRIs.

Explicit masking provides useful features, like replacing password instead of removing.
(For example, it is possible to write `http://user:password@example.com` as `http://user:${password}@example.com` or `http://user:********@example.com`.)
